app/src/main/java/com/example/myanimereport/fragments/ReportFragment.java

```diff
@@ -24,7 +72,316 @@ public View onCreateView(@NonNull LayoutInflater inflater,
+    private void setTop5() {
+        // Sort by rating (descending) and get the top 5
+        List<Entry> sortedEntries = new ArrayList<>(entries);
+        sortedEntries.sort((e1, e2) -> Double.compare(e2.getRating(), e1.getRating()));
+        if (sortedEntries.size() > 5) sortedEntries = sortedEntries.subList(0, 5);
+
+        // Set up adapter and layout of recycler view
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
+        EntriesAdapter adapter = new EntriesAdapter(this, sortedEntries, false);
+        binding.rvTop5.setLayoutManager(layoutManager);
+        binding.rvTop5.setAdapter(adapter);
+    }
+
+    /* Calculates the average rating of a list of entries. */
+    public double getAverageRating(List<Entry> entries) {
+        double sumRating = 0;
+        for (Entry entry : entries) sumRating += entry.getRating();
+        return sumRating / entries.size();
+    }
+
+    /* Calculates and sets the percent difference between two numbers. */
+    public void setComparisonText(double prev, double curr, TextView tvCompare) {
+        double diff = (curr - prev) / prev;
+        if (diff > 0) {
+            tvCompare.setText(String.format(Locale.getDefault(), "\u2191 %.1f%%", diff * 100));
+            tvCompare.setTextColor(green);
+        } else if (diff == 0) {
+            tvCompare.setText(String.format(Locale.getDefault(), "\u2195 %.1f%%", diff * 100));
+            tvCompare.setTextColor(white);
+        } else if (diff < 0) {
+            tvCompare.setText(String.format(Locale.getDefault(), "\u2193 %.1f%%", diff * 100));
+            tvCompare.setTextColor(red);
+        }
+    }
+
+    /* Refreshes a chart. */
+    public void refreshChart(Chart chart) {
+        chart.animateXY(2000,2000);
+        chart.invalidate();
+    }
+
+    /* General styling of the x-axis. */
+    public void customizeXAxis(XAxis xAxis) {
+        xAxis.setTextColor(white);
+        xAxis.setPosition(XAxis.XAxisPosition.BOTTOM);
+        xAxis.setGridLineWidth(1f);
+        xAxis.setGranularity(1f);
+    }
+
+    /* General styling of the main y-axis. */
+    public void customizeYAxisMain(YAxis yAxis) {
+        yAxis.setGridLineWidth(1f);
+        yAxis.setTextColor(white);
+    }
+
+    /* General styling of the secondary y-axis. */
+    public void customizeYAxisSecondary(YAxis yAxis) {
+        yAxis.setDrawLabels(false);
+        yAxis.setDrawGridLines(false);
+    }
+
+    /* General styling of a chart. Mode is for different ways of drawing the marker view. */
+    public void customizeChart(Chart chart, int mode) {
+        chart.getDescription().setEnabled(false);
+        chart.getLegend().setTextColor(white);
+        chart.getLegend().setWordWrapEnabled(true);
+        chart.getLegend().setYOffset(3f);
+        CustomMarkerView mv = new CustomMarkerView(getContext(), R.layout.custom_marker_view_layout, mode);
+        mv.setChartView(chart);
+        chart.setMarker(mv);
```

Comment:
nit: This file is getting to be quite long. I know you have a lot of helper functions here. Is there any way to break this file down a bit more?